benchmark/result/result.py

```diff
@@ -0,0 +1,51 @@
+"""Basic result classes for benchmarking."""
+
+from typing import Any
+from uuid import UUID
+
+from attrs import define, field
+from pandas import DataFrame
+
+
+@define(frozen=True)
+class Result:
+    """A single result of the benchmarking."""
+
+    name: str = field()
+    """The name of the benchmarking result."""
+
+    identifier: UUID = field()
+    """The unique identifier of the benchmark running which can be set
+    to compare different executions of the same benchmark setting."""
+
+    metadata: dict[str, str] = field()
+    """Metadata about the benchmarking result."""
+
+    benchmark_result: DataFrame = field()
```

Comment:
@Scienfitz, @AVHopp: I'm not yet sure about this part. Do we actually want to restrict us to dataframes only? From the benchmark perspective, the classes probably don't need to be hardcoded to Dataframes, and we haven't yet really fully decided if there are gonna be "other" benchmarks (i.e. such that don't just return a convergence dataframe but, for instance, just some a dict, an image, whatever, ...). Opinions? Perhaps a topic for the dev meeting

Reply:
Can you provide a specific example of a reasonable benchmark whose result would not be a dataframe resp not something derived from a dataframe (like an image)? I think having a dataframe is already quite general and I do not see other things that should be returned and checked automatically.
Also, to use your own argument 😉  When we figure out that there is indeed something else that we want to have as a result of a benchmark, we can still add it later. Since we will never decide "Dataframes are no longer valid results from benchmarks", there should also be only little problems if we ever decide to do that.

Reply:
I could imagine stuff a simple as returning a dict or just a float. But it's fine, let's generalize once needed. Resolving